helper/schema/provider.go

```diff
@@ -284,7 +284,7 @@ func (p *Provider) Apply(
 		return nil, fmt.Errorf("unknown resource type: %s", info.Type)
 	}
 
-	return r.Apply(s, d, p.meta)
+	return r.Apply(nil, s, d, p.meta)
```

Comment:
If this is dead code, can we remove it?

Reply:
It is dead from protocol 5's point of view but deleting the func will mean `schema.Provider` won't implement `terraform.ResourceProvider` and somewhere in `plugin.Serve` we are expecting an implementation of `terraform.ResourceProvider` I think it gets cast to `schema.Provider` right away though, I think the interface was only needed in V1.... I will investigate and at the very least this should have a `panic("should never be called")` good catch.